models/surveys.go

```diff
@@ -87,9 +100,66 @@ func NewAPI(db *sql.DB) (*API, error) {
 		GetSurveyIDStmt:                   getSurveyIDStmt,
 		GetClassifierTypeSelectorStmt:     getClassifierTypeSelectorStmt,
 		GetClassifierTypeSelectorByIDStmt: getClassifierTypeSelectorByIDStmt,
+		GetSurveyRefStmt:                  getSurveyRefStmt,
+		PutSurveyDetailsBySurveyRefStmt:   putSurveyDetailsBySurveyRefStmt,
 	}, nil
 }
 
+// PutSurveyDetails endpoint handler changes a survey short name using the survey reference
+func (api *API) PutSurveyDetails(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	surveyRef := vars["ref"]
+
+	body, err := ioutil.ReadAll(r.Body)
+
+	type Data struct {
+		SurveyID   string `json: "surveyId"`
+		SurveyRef  string `json: "surveyRef"`
+		ShortName  string `json: "shortName"`
+		LongName   string `json: "longName"`
+		LegalBasis string `json: "legalBasis"`
+	}
+	var putData Data
+	err = json.Unmarshal(body, &putData)
+	if err != nil {
+		http.Error(w, "Error unmarshalling JSON", http.StatusBadRequest)
+	}
+
+	shortName := putData.ShortName
+	longName := putData.LongName
+
+	err = api.getSurveyRef(surveyRef)
```

Comment:
Why make an additional database request to validate the survey exists? Could you not just attempt the update and return 404 if no rows were updated?

Reply:
Could do was more consistent with other endpoints but that would make sense.